src/jvm/main/org/jetbrains/kotlinx/lincheck/transformation/LincheckJavaAgent.kt

```diff
@@ -312,17 +325,21 @@ internal object LincheckClassFileTransformer : ClassFileTransformer {
     override fun transform(
         loader: ClassLoader?, className: String, classBeingRedefined: Class<*>?, protectionDomain: ProtectionDomain?, classBytes: ByteArray
     ): ByteArray? = runInIgnoredSection {
-        if (instrumentationMode == MODEL_CHECKING && !INSTRUMENT_ALL_CLASSES_IN_MODEL_CHECKING_MODE) {
-            // In the model checking mode, we transform classes lazily,
-            // once they are used in the testing code.
-            if (className.canonicalClassName !in instrumentedClassesInTheModelCheckingMode) return null
-        } else {
-            if (!shouldTransform(className.canonicalClassName, instrumentationMode)) return null
+        // If the class should not be transformed, return immediately.
+        if (!shouldTransform(className.canonicalClassName, instrumentationMode)) {
+            return null
+        }
+        // In the model checking mode, we transform classes lazily,
+        // once they are used in the testing code.
+        if (!INSTRUMENT_ALL_CLASSES &&
+            instrumentationMode == MODEL_CHECKING &&
+            className.canonicalClassName !in instrumentedClasses) {
+            return null
         }
         return transformImpl(loader, className, classBytes)
     }
 
-    private fun transformImpl(loader: ClassLoader?, className: String, classBytes: ByteArray): ByteArray = transformedClassesCache.computeIfAbsent(className) {
+    private fun transformImpl(loader: ClassLoader?, className: String, classBytes: ByteArray): ByteArray = transformedClassesCache.computeIfAbsent(className.canonicalClassName) {
```

Comment:
Why `canonicalClassName`?

Reply:
Because the other place where I use `transformedClassesStress` is inside `install` method, where I have the `Class` instance, and I can only obtain its canonical name. 

So it is easier to convert ASM's internal name to the canonical name here, than the other way around.

Reply:
Alright, I got it. Can we please use `canonicalClassName` and `internalClassName` everywhere to highlight whether the class name is canonical or internal in each case? I understand that it is not a part of this PR, but with this explicit notion, it is easier to review the PR properly.

Reply:
Ok, I renamed to explicit `canonicalClassName` or `internalClassName` everywhere in the `LincheckJavaAgent.kt`.